src/app/components/elements/EventBookingForm.tsx

```diff
@@ -40,273 +249,93 @@ export const EventBookingForm = ({
           </legend>
           <RS.Row>
             <RS.Col md={6}>
-              {/* Should be impossible to not have a first name */}
-              <RS.Label htmlFor="account-firstname" className="form-required">
-                First name
-              </RS.Label>
-              <RS.Input
-                id="account-firstname"
-                name="firstname"
-                type="text"
-                disabled
-                value={targetUser.givenName || ""}
-              />
+              <DisabledInputWithLabel type="firstname" />
             </RS.Col>
             <RS.Col md={6}>
-              {/* Should be impossible to not have a last name */}
-              <RS.Label htmlFor="account-lastname" className="form-required">
-                Last name
-              </RS.Label>
-              <RS.Input
-                id="account-lastname"
-                name="secondname"
-                type="text"
-                disabled
-                value={targetUser.familyName || ""}
-              />
+              <DisabledInputWithLabel type="lastname" />
             </RS.Col>
           </RS.Row>
 
           <div>
             <RS.Row>
-              <RS.Col md={6}>
-                <RS.Label htmlFor="account-email" className="form-required">
-                  Email address
-                </RS.Label>
-                <RS.Input
-                  id="account-email"
-                  name="email"
-                  type="email"
-                  disabled
-                  value={targetUser.email || ""}
-                  invalid={targetUser.emailVerificationStatus != "VERIFIED"}
-                />
-                <RS.FormFeedback>
-                  {editingSelf
-                    ? "You must verify your email address to book on events. This is so we can send you details about the event."
-                    : "WARNING: This email is not verified. The details about the event might not reach the user."}
-                </RS.FormFeedback>
+              <RS.Col md={12}>
+                <DisabledInputWithLabel type="email" invalid={targetUser.emailVerificationStatus !== "VERIFIED"} />
+                {targetUser.emailVerificationStatus != "VERIFIED" && <EmailVerification />}
```

Comment:
Given that you seem to be using `targetUser.emailVerificationStatus !== "VERIFIED"` in a few places, you could have saved the result of that in a const with a meaningful name, like `notVerifiedEmail` or similar.